src/Microsoft.AspNet.Razor/Generator/Compiler/CodeBuilder/CSharp/CSharpTagHelperCodeRenderer.cs

```diff
@@ -448,7 +448,7 @@ private void RenderTagOutput(string tagOutputMethodName)
                 return;
             }
 
-            CSharpCodeVisitor.RenderPreWriteStart(_writer, _context);
+            CSharpCodeVisitor.RenderPreWriteStart(_writer, _context, isTagHelperOutput: true);
```

Comment:
This method is used evvverrrywhere, just make a separate helper, don't overrite the existing impl.